.github/workflows/quality.yml

```diff
@@ -0,0 +1,61 @@
+name: Java Code Quality Analysis
+
+on:
+  push:
+    branches: [ develop ]
+  pull_request:
+    types: [ opened, synchronize, reopened ]
+
+jobs:
+  code-quality-analysis:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Set up Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: '3.13'
+
+      - name: Check LOC and number of class dependencies
+        run: |
+          # Run the analysis and capture output
+          output=$(python supporting_scripts/analyze_java_files.py \
+            --max-lines ${{ vars.MAX_LINES || 1000 }} \
+            --max-params ${{ vars.MAX_PARAMS || 10 }} \
+            ${{ vars.INCLUDE_REPO_DEPS == 'true' && '--include-repo-deps' || '' }})
```

Comment:
Is there any particular reason for adding the option to specify this via GitHub vars? 
Also, if you want to keep them, please give them more descriptive names. The current ones do not indicate what their purpose is when just reading their names in the GitHub vars list.